assets/javascripts/obs_rsync.js

```diff
@@ -0,0 +1,38 @@
+function postAndRefresh(btn, delay){
+    var post_url = $(btn).attr("post_url");
+    $.ajax({
+        url: post_url,
+        method: 'POST',
+        dataType: 'json',
+        success: function(data) {
+            if (delay) {
+                setTimeout(function() {
+                    location.reload();
+                }, delay);
+            } else {
+                location.reload();
+            }
+        },
+        error: function(xhr, ajaxOptions, thrownError) {
+            var message = xhr.responseJSON.error;
+            addFlash('danger', 'Error: ' + message);
```

Comment:
`xhr.responseJSON.error` might be `undefined` here (and in the error handler of the other function).